packages/ember-runtime/lib/system/object.js

```diff
@@ -21,7 +24,48 @@ import { DEBUG } from 'ember-env-flags';
   @uses Ember.Observable
   @public
 */
-const EmberObject = CoreObject.extend(Observable);
+const EmberObject = CoreObject.extend(Observable, {
+  _debugContainerKey: descriptor({
+    enumerable: false,
+    get() {
+      if (this[OVERRIDE_CONTAINER_KEY]) {
+        return this[OVERRIDE_CONTAINER_KEY];
+      }
+
+      let meta = peekMeta(this);
+      let factory  = meta.getFactory();
```

Comment:
if we `peekMeta` `meta` may be `undefined` so we must guard before accessing `getFactory`

something like:

```js
let meta = peekMeta(this)
if (meta === undefined) { return; }
let factory = meta.getFactory();
if (factory === undefined) { return ; }
return factory.fullName;
```